### PR TITLE
Add sonar pr decoration

### PR DESCRIPTION
**Pull request by npalm**

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -4,13 +4,16 @@ A Github action for running static analysis on your codebase using SonarQube ins
...
-| token       | SonarQube login token                                      |
-| url         | SonarQube server url                                       |
+| Parameter                   | Description                                                              | required | default |
+| --------------------------- | :----------------------------------------------------------------------- | -------- | ------- |
+| projectName                 | Your application's name in SonarQube                                     | yes      |         |
+| projectKey                  | Unique value for your project in SonarQube                               | yes      |         |
+| baseDir                     | Relative path to your sonar.properties file (default: `.`)               | no       | `.`     |
+| token                       | SonarQube login token                                                    | yes      |         |
+| url                         | SonarQube server url                                                     | yes      |         |
+| scmPovider                  | SCM provider                                                             | no       | `git`   |
```
Comment (PatrickMennen) [inline README.MD]: scmProvider -> typo

**Comment by npalm** [comment]
@PatrickMennen thanks for merging the two test files. But now the tests are not the same anymore. All tests are using now a context with the pull request payload.

File: src/sonarScanner-p.test.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { sonarScanner } from './sonarScanner';
+import { exec } from '@actions/exec';
+
+jest.mock('@actions/exec');
+
+jest.mock('@actions/github', () => ({
```
Comment (PatrickMennen) [inline src/sonarScanner-p.test.ts:6]: isn't this mock obsolete?

**Review by PatrickMennen — COMMENTED** [review]
Your tests are failing at the moment

File: src/sonarScanner.ts [hunk 3]
```diff
@@ -7,27 +8,68 @@ export const sonarScanner = async () => {
...
-  core.info(`
-    Using Configuration:
-
-    ProjectName: ${projectName}
-    ProjectKey : ${projectKey}
-    BaseDir    : ${baseDir}
-    Token      : ${token}
-    URL        : ${url}
-  `);
+  function getBranchOrTagName(githubRef: string): string {
```
Comment (PatrickMennen) [inline src/sonarScanner.ts]: move outside of the function scope

**Review by PatrickMennen — CHANGES_REQUESTED** [review]
getBranchOrTagName should not be inside the const and your unittests are failing at the moment.

**Review by PatrickMennen — APPROVED** [review]
Ok now!

**Pull request merged**